src/MiniCover/HitServices/HitServiceWithTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace MiniCover
+{
+    public static class HitServiceWithTests
+    {
+        private static readonly object lockObject = new object();
+        private static Dictionary<string, Dictionary<int, HitTrace>> files = new Dictionary<string, Dictionary<int, HitTrace>>();
+
+        public static void Init(string fileName)
+        {
+            lock (lockObject)
+            {
+
+                if (!files.ContainsKey(fileName))
+                {
+                    files[fileName] = new Dictionary<int, HitTrace>();
+                }
+            }
+        }
+
+        public static void Hit(string fileName, int id)
+        {
+            lock (lockObject)
+            {
+                StackTrace stackTrace = new System.Diagnostics.StackTrace();
+
+                var currentIndex = 0;
+
+                var frames = stackTrace.GetFrames();
+                foreach (var currentFrame in frames)
+                {
+                    if (currentFrame.GetMethod().Name == "InvokeMethod")
```

Comment:
I'm afraid this is not a guaranteed way to find the test method. I think it would fail if:

- Anyone creates a method called InvokeMethod, it would break the chain
- Anyone call InvokeMethod from .NET framework
- A specific test runner do not call InvokeMethod to run the tests

Do you think it would be too heavy if we save the whole call stack and show it on reports?

Reply:
Maybe we can change instrumentation to something like:
```csharp
void MyMethod() {
   var method = HitService.EnterMethod();
   method.Hit("coverage.hits", 1);
   //instruction 1
   method.Hit("coverage.hits", 2);
   //instruction 2
}
```
Then we record the call stack only when it enters on a new method, and we also correlate all hits inside that method with that call stack.

Reply:
My way to do it was a quick win,
I know about other calling InvokeMethod.
 I still think using stack trace in hit service is the best like that we don't impact the fast way with MinimalHitService.

But I will try to see a best way.